windows-gaming-driver/src/setup/vfio.rs

```diff
@@ -82,82 +82,69 @@ pub fn setup(setup: &mut SetupConfig, machine: &mut MachineConfig) -> bool {
+        println!("You will not be able to use them on your host system while Windows is running.");
         println!();
-        println!("U done goofed m8.");
-        println!("Ur on ur own :)");
+
+        while select_device(machine, "Which device woud you like to pass through?", &pci_devs,
+                            |x| x.resettable)? { }
+
+        println!("Note that non-resettable passthrough is only known to work for GPUs.");
+        println!("With everything else you're completely on your own.");
+        println!("You will not be able to use these devices on your host system at all.");
         println!();
-        let all_devices = pci_devs.iter().filter(|x| x.pci_class != "60400").collect();
-        while let Ok(Some(())) = select_device(setup, machine, "Which device do you want to pass through?", &all_devices) {};
+        if ask::yesno("Would you like to pass any non-resettable PCI devices=") {
+            // all non-resettable devices but no PCI bridges (because arch wiki says so)
+            while select_device(machine, "Which device do you want to pass through?", &pci_devs,
+                                |x| !x.resettable && x.pci_class != "60400")? { }
+        }
     }
-    Ok(())	
+    Ok(())
 }
 
-fn select_device(setup: &mut SetupConfig, machine: &mut MachineConfig, question: &str, devices: &Vec<&PciDevice>) -> Result<Option<()>>{
-    let udev = Context::new().expect("Failed to create udev context");
-    let mut iter = Enumerator::new(&udev)?;
-    iter.match_subsystem("pci")?;
-    let udev_devices = iter.scan_devices()?.map(PciDevice::new);
-    let pci_devs: Vec<_> = udev_devices.filter(|x| !machine.vfio_slots.iter().any(|y| y.device == x.id)).collect();
-    let askable_devices: Vec<_> = devices.iter().filter(|x| !machine.vfio_slots.iter().any(|y| y.device == x.id)).collect();
-    
+fn select_device<P: Fn(&PciDevice) -> bool>(machine: &mut MachineConfig,
+                                            question: &str,
+                                            devices: &[PciDevice],
+                                            predicate: P) -> Result<bool> {
+    let askable_devices: Vec<_> = devices.iter()
+        .filter(|x| predicate(x) && !machine.pci_devices.iter().any(|y| y.device == x.id)).collect();
+
     println!("{}", devices.len());
-    
+
     for (i, dev) in askable_devices.iter().enumerate() {
         println!("[{}]\t{}", i, dev);
     }
     println!("[{}]\tNone of the above.", askable_devices.len());
 
     let selection = ask::numeric(question, 0..askable_devices.len()+1);
-    
-    if selection < askable_devices.len() {
-        let selected = &askable_devices[selection];
-        for device in pci_devs.iter().filter(|x| x.pci_device() == selected.pci_device()) {
+
+    if let Some(selected_id) = askable_devices.get(selection).map(|x| x.id) {
+        for device in devices.iter().filter(|x| x.id == selected_id) {
```

Comment:
This does not do what you think it does. This check for the specific id. However the graphics and audio part of a gpu have different ids: 
> NVIDIA Corporation Unknown model [10de:1b06]
> NVIDIA Corporation GP102 HDMI Audio Controller [10de:10ef]